diff --git a/ios/RNViewShot.mm b/ios/RNViewShot.mm
index 2cf632fb8dfb09aba637f8c8700fbe02b926404c..8c8eb8ab30f06582a15d1c4253e77059c02e32be 100644
--- a/ios/RNViewShot.mm
+++ b/ios/RNViewShot.mm
@@ -80,12 +80,11 @@ RCT_EXPORT_METHOD(captureRef:(nonnull NSNumber *)target
     UIView* rendered;
     UIScrollView* scrollView;
     if (snapshotContentContainer) {
-      if (![view isKindOfClass:[RCTScrollView class]]) {
-        reject(RCTErrorUnspecified, [NSString stringWithFormat:@"snapshotContentContainer can only be used on a RCTScrollView. instead got: %@", view], nil);
+      if (![view isKindOfClass:[UIScrollView class]]) {
+        reject(RCTErrorUnspecified, [NSString stringWithFormat:@"snapshotContentContainer can only be used on a ScrollView. instead got: %@", view], nil);
         return;
       }
-      RCTScrollView* rctScrollView = view;
-      scrollView = rctScrollView.scrollView;
+      scrollView = (UIScrollView*)view;
       rendered = scrollView;
     }
     else {
