diff --git a/dist/types/index.d.ts b/dist/types/index.d.ts
index 6a73d1ca857561efd5eebdc0dc13f8c5c124e25d..535af43d4d0d94a40769292c493119661181021e 100644
--- a/dist/types/index.d.ts
+++ b/dist/types/index.d.ts
@@ -1 +1,7 @@
+declare module '@hugeicons/core-free-icons/*' {
+  const icon:
+    | [string, { [key: string]: string | number }][]
+    | readonly (readonly [string, { readonly [key: string]: string | number }])[];
+  export default icon;
+}
 export { default as HugeiconsIcon, HugeiconsIconProps, HugeiconsProps, IconSvgElement } from './HugeiconsIcon.js';
