diff --git a/node_modules/@hugeicons/react/dist/types/core-free-icons.d.ts b/node_modules/@hugeicons/react/dist/types/core-free-icons.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..42cb2358eb60a994a7475b7b0cb07ad80209f1dc
--- /dev/null
+++ b/node_modules/@hugeicons/react/dist/types/core-free-icons.d.ts
@@ -0,0 +1,4 @@
+declare module '@hugeicons/core-free-icons/*' {
+  const icon: import('./HugeiconsIcon').IconSvgElement;
+  export default icon;
+}
diff --git a/node_modules/@hugeicons/react/dist/types/index.d.ts b/node_modules/@hugeicons/react/dist/types/index.d.ts
index 6a73d1ca857561efd5eebdc0dc13f8c5c124e25d..54ee6e08e7d3be297a28340dd8f2b9d6034b28a3 100644
--- a/node_modules/@hugeicons/react/dist/types/index.d.ts
+++ b/node_modules/@hugeicons/react/dist/types/index.d.ts
@@ -1 +1,2 @@
+/// <reference path="./core-free-icons.d.ts" />
 export { default as HugeiconsIcon, HugeiconsIconProps, HugeiconsProps, IconSvgElement } from './HugeiconsIcon.js';
