diff --git a/node_modules/@convex-dev/better-auth/dist/plugins/cross-domain/client.js b/node_modules/@convex-dev/better-auth/dist/plugins/cross-domain/client.js
index ae0c621..b2d0788 100644
--- a/node_modules/@convex-dev/better-auth/dist/plugins/cross-domain/client.js
+++ b/node_modules/@convex-dev/better-auth/dist/plugins/cross-domain/client.js
@@ -52,7 +52,7 @@ export function getCookie(cookie) {
         // noop
     }
     const toSend = Object.entries(parsed).reduce((acc, [key, value]) => {
-        if (value.expires && value.expires < new Date()) {
+        if (value.expires && new Date(value.expires) < new Date()) {
             return acc;
         }
         return `${acc}; ${key}=${value.value}`;
@@ -140,6 +140,9 @@ export const crossDomainClient = (opts = {}) => {
                             !opts?.disableCache) {
                             const data = context.data;
                             storage.setItem(localCacheName, JSON.stringify(data));
+                            if (data === null) {
+                                storage.setItem(cookieName, "{}");
+                            }
                         }
                     },
                 },
@@ -165,7 +168,7 @@ export const crossDomainClient = (opts = {}) => {
                             error: null,
                             isPending: false,
                         });
-                        storage.setItem(localCacheName, "{}");
+                        storage.setItem(localCacheName, "null");
                     }
                     return {
                         url,
diff --git a/node_modules/@convex-dev/better-auth/dist/react/index.js b/node_modules/@convex-dev/better-auth/dist/react/index.js
index eb292f6..235eef4 100644
--- a/node_modules/@convex-dev/better-auth/dist/react/index.js
+++ b/node_modules/@convex-dev/better-auth/dist/react/index.js
@@ -73,7 +73,7 @@ function useUseAuthFromBetterAuth(authClient, initialToken) {
         [sessionId]);
         return useMemo(() => ({
             isLoading: isSessionPending,
-            isAuthenticated: session !== null,
+            isAuthenticated: Boolean(session?.session),
             fetchAccessToken,
         }), 
         // eslint-disable-next-line react-hooks/exhaustive-deps
