diff --git a/templates/start-app/src/routes/__root.tsx b/templates/start-app/src/routes/__root.tsx
index c6988c007e4..ea8b325c4c0 100644
--- a/templates/start-app/src/routes/__root.tsx
+++ b/templates/start-app/src/routes/__root.tsx
@@ -25,6 +25,12 @@ export const Route = createRootRoute({
       },
     ],
   }),
+  notFoundComponent: () => (
+    <main className="container mx-auto p-4 pt-16">
+      <h1>404</h1>
+      <p>The requested page could not be found.</p>
+    </main>
+  ),
   shellComponent: RootDocument,
 })
 
diff --git a/templates/start-monorepo/apps/web/src/routes/__root.tsx b/templates/start-monorepo/apps/web/src/routes/__root.tsx
index 259f4a49ee0..33e2a8c107a 100644
--- a/templates/start-monorepo/apps/web/src/routes/__root.tsx
+++ b/templates/start-monorepo/apps/web/src/routes/__root.tsx
@@ -23,6 +23,12 @@ export const Route = createRootRoute({
       },
     ],
   }),
+  notFoundComponent: () => (
+    <main className="container mx-auto p-4 pt-16">
+      <h1>404</h1>
+      <p>The requested page could not be found.</p>
+    </main>
+  ),
   shellComponent: RootDocument,
 })
 
